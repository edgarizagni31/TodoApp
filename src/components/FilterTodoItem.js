const FilterTodoItem = ( { text, fn  }  ) => {
    return (
        <p 
          className= 'todo-list__option'
          onClick = { fn }
        >
            { text }
        </p>

    )
}

export default FilterTodoItem;
