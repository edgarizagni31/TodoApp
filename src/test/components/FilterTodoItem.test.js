import FilterTodoItem from '../../components/FilterTodoItem';
import { shallow } from 'enzyme';

describe('test to FilterTodoItem', () => {
    const text = 'Activas';
    const filterComplete = jest.fn();
    const wrapper = shallow(<FilterTodoItem  text = {text} fn = { filterComplete } />);

    test('should displayed be correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('button').text().trim()).toBe(text);
    });

    test('should call fn when click', () => {
        wrapper.find('button').simulate('click');
        expect( filterComplete ).toHaveBeenCalledTimes(1);
    });
});
