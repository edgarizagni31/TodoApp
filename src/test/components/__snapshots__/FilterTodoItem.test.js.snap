// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to FilterTodoItem should displayed be correctly 1`] = `
<Fragment>
  <button
    className="todo-list__option"
    onClick={[MockFunction]}
  >
    Activas
  </button>
</Fragment>
`;
